pkgs/development/interpreters/jruby/default.nix

```diff
@@ -1,63 +1,26 @@
-{ stdenv, callPackage, fetchurl, makeWrapper, jre }:
+{ callPackage, lib }:
+rec {
 
-let
-# The version number here is whatever is reported by the RUBY_VERSION string
-rubyVersion = callPackage ../ruby/ruby-version.nix {} "2" "5" "7" "";
-jruby = stdenv.mkDerivation rec {
-  pname = "jruby";
+  rubyVersion = import ../ruby/ruby-version.nix { inherit lib; };
 
-  version = "9.2.13.0";
-
-  src = fetchurl {
-    url = "https://s3.amazonaws.com/jruby.org/downloads/${version}/jruby-bin-${version}.tar.gz";
-    sha256 = "0n5glz6xm3skrfihzn3g5awdxpjsqn2k8k46gv449rk2l50w5a3k";
+  generic_jruby =  {version, sha256, rubyVersion}: callPackage ./generic.nix {
+    inherit version;
+    inherit sha256;
+    inherit rubyVersion;
   };
 
-  buildInputs = [ makeWrapper ];
-
-  installPhase = ''
-     mkdir -pv $out/docs
-     mv * $out
-     rm $out/bin/*.{bat,dll,exe,sh}
-     mv $out/COPYING $out/LICENSE* $out/docs
-
-     for i in $out/bin/jruby{,.bash}; do
-       wrapProgram $i \
-         --set JAVA_HOME ${jre}
-     done
-
-     ln -s $out/bin/jruby $out/bin/ruby
-
-     # Bundler tries to create this directory
-     mkdir -pv $out/${passthru.gemPath}
-     mkdir -p $out/nix-support
-     cat > $out/nix-support/setup-hook <<EOF
-       addGemPath() {
-         addToSearchPath GEM_PATH \$1/${passthru.gemPath}
-       }
-
-       addEnvHooks "$hostOffset" addGemPath
-     EOF
-  '';
-
-  passthru = rec {
-    rubyEngine = "jruby";
-    gemPath = "lib/${rubyEngine}/gems/${rubyVersion.libDir}";
-    libPath = "lib/${rubyEngine}/${rubyVersion.libDir}";
+  v9_2_12_0 = generic_jruby {
+    version = "9.2.12.0";
+    sha256 = "013c1q1n525y9ghp369z1jayivm9bw8c1x0g5lz7479hqhj62zrh";
+    rubyVersion = rubyVersion "2" "5" "7" "";
   };
 
-  meta = with stdenv.lib; {
-    description = "Ruby interpreter written in Java";
-    homepage = "http://jruby.org/";
-    license = with licenses; [ cpl10 gpl2 lgpl21 ];
-    platforms = platforms.unix;
-    maintainers = [ maintainers.fzakaria ];
-  };
-};
-in jruby.overrideAttrs (oldAttrs: {
-  passthru = oldAttrs.passthru // {
-    devEnv = callPackage ../ruby/dev.nix {
-      ruby = jruby;
-    };
```

Comment:
This is unnecessary and inlined within the generic builder.
It relies on "self" to be self-referential to the ruby it's building.